tests/unit/staff/views/test_projects.py

```diff
@@ -421,6 +422,7 @@ def test_projectedit_post_success_when_not_changing_slug(rf, core_developer):
     assert project.slug == "test"
     assert project.number == 456
     assert project.copilot == new_copilot
+    assert project.redirects.count() == 0
```

Comment:
It's a bit of a drive-by I know but I thought it was a clear omission that the previous tests tested that a redirect was created in the expected circumstances but not that one wasn't created when the criteria were not met.

Reply:
I'm failing to understand why this is an overreach then [why is this line not](https://github.com/opensafely-core/job-server/blob/0f265af7f0017a9ca39933c1e6d1b4b60f7e71fd/tests/unit/staff/views/test_projects.py#L392)?

There's a pre-existing test for "update project attributes inclusive of slug" where the expected result is that those attributes are updated and a corresponding redirect is created. There's a pre-existing test for "update project attributes exclusive of slug" (this one) and the expected result is that those attributes are updated but we do not also check the redirects here for some reason?

If we're testing that we're calling `projects.edit` with the right arguments then if we are inspecting the results of the expected actions of `projects.edit` - as the pre-existing tests do - then these tests are already reaching beyond just checking the arguments AFAICT.

Reply:
I think you've made a good case for that line also being an overreach! Alas, I didn't consider that line, because it's not touched by this PR. I'll think about this more, but it's not a deal-breaker. I'm more interested in future us being able to parse the comparison.